### PR TITLE
Fix android.resource Uris that point to a vector drawable fail to load

### PR DESCRIPTION
**Pull request by pkpdeveloper**
Hi @colinrtwhite,

Please review and merge fix for [android.resource Uri](https://github.com/coil-kt/coil/issues/42) related issue. also added test cases for that.
I have tried many solutions only this one works as expected.

Please let me know for further improvements or any discussion. 

Cheers :beers:  for creating this awesome library :clap: :heart:

### CODE REVIEW

File: coil-base/src/main/java/coil/fetch/UriFetcher.kt [hunk 1]
```diff
@@ -42,6 +44,18 @@ internal class UriFetcher(
         size: Size,
         options: Options
     ): FetchResult {
+        if (data.scheme == ContentResolver.SCHEME_ANDROID_RESOURCE) {
+            val resourceId = extractResourceId(data)
+            resourceId?.let {
+                val drawable = context.getDrawableCompat(resourceId)
```
Comment (audkar) [inline coil-base/src/main/java/coil/fetch/UriFetcher.kt]: If I understand correctly, this code part assumes that `resourceId` belongs to same applications. However `Uri` can point to `resourceId` which belongs to different application:
```
"android.resource://{pkgName}/{resourceId}"
```

Comment (pkpdeveloper) [reply coil-base/src/main/java/coil/fetch/UriFetcher.kt]: Thanks, @audkar for pointing this issue. I got the solution from [Glide source code](https://github.com/bumptech/glide/blob/d8278f43a94484f20822170b71ac7b66269a2aa6/library/src/main/java/com/bumptech/glide/load/resource/drawable/ResourceDrawableDecoder.java) 
I will do the changes ASAP :blush:

**Comment by audkar** [comment]
Implementation logic seems correct and aligned with platform one (private [method](https://android.googlesource.com/platform/frameworks/base/+/9be54d400d68c735013bc8069fbcb66c3f98c3ee/core/java/android/content/ContentResolver.java#470) used for `ImageView.setImageURI(uri)`). Handles boths possible formats for resource uri. Good job

**Comment by colinrtwhite** [comment]
Thanks for the contribution. We're working on a bigger refactor of the mappers + fetchers [here](https://github.com/coil-kt/coil/pull/32), which might conflict with this PR. I'll circle back and review this PR after the one I linked is merged.

**Pull request closed without merging**